File: portfolioTracker/lib/models/transaction_model.dart
class Transaction {
  String date;
  double amount;
  String opponent;
  String opponentAccount;
  String comment;
  String ownAccount;

  Transaction(
      {this.date,
      this.amount,
      this.opponent,
      this.opponentAccount,
      this.comment,
      this.ownAccount});

  factory Transaction.fromJSON(Map<String, dynamic> json) {
    return Transaction(
        date: json['date'],
        amount: json['amount'],
        opponent: json['opponent'],
        opponentAccount: json['opponent_account'],
        comment: json['comment'],
        ownAccount: json['own_account']);
  }
}

File: backend/app/models/transaction_model.py
from pydantic import BaseModel
import datetime


class TransactionBase(BaseModel):
    date: datetime.date
    amount: float
    opponent: str
    opponent_account: str
    comment: str
    own_account: str
    category_id: int


class TransactionInDB(TransactionBase):
    id: int

    class Config:
        orm_mode = True
